sdk/keyvault/azure-keyvault-certificates/azure/keyvault/certificates/_shared/async_client_base.py

```diff
@@ -104,3 +104,13 @@ def _build_pipeline(config: Configuration, transport: AsyncHttpTransport, **kwar
     @property
     def vault_url(self) -> str:
         return self._vault_url
+
+    async def __aenter__(self) -> "AsyncKeyVaultClientBase":
+        await self._client._pipeline.__aenter__()
```

Comment:
This class should wrap the generated client's aenter/exit, not the aenter/exit of a protected attribute.